### ISSUE TITLE
Report progress callback

### ISSUE THREAD

**Issue by AndreMiras** [makinacorpus/landez#48]
I'm integrating your library to a [Kivy](https://github.com/kivy/kivy) graphical application.
I would like the user to know roughly the progress of the process.
I guess this could be achieved with a callback we could pass to the ```MBTilesBuilder.run()``` method.

**Comment by leplatrem** [comment]
Yes, why not. Or you could also poll a class attribute at regular interval. As you prefer...

**Comment by AndreMiras** [comment]
Thanks @leplatrem, I had a look at which attribute I could poll and I found only two relevant ones.
```MBTilesBuilder.nbtiles``` and ```TilesManager.rendered```.
While ```MBTilesBuilder.nbtiles``` gives the number of tiles in total for a given bbox, ```TilesManager.rendered``` gives the number of tiles that were downloaded. The problem is that when accessing cached tiles, the ```rendered``` counter doesn't get incremented. Which means we do not always get TilesManager.rendered = MBTilesBuilder.nbtiles at the end.
I think I'll probably use your polling method even though I'm not a big fan of polling patterns (compared to callbacks), but I'll inherit ```MBTilesBuilder``` to specialise it and always increment a counter on each loop.

**Comment by leplatrem** [comment]
As you wish !
Don't hesitate to provide a pull-request to introduce your callback ! We'll review it with pleasure ! Especially if it comes with a small test case.
Thanks for your feedback !
